recipes/catch2/2.x.x/conanfile.py

```diff
@@ -46,7 +46,7 @@ def configure(self):
             self.options.rm_safe("with_benchmark")
 
     def package_id(self):
-        if not self.options.with_main:
+        if not self.info.options.with_main:
```

Comment:
Probably not worth creating new RREV for something which is unclear for the moment and might be reverted for conan v2: https://github.com/conan-io/conan/issues/12693

Reply:
😉 that's why we are looking at some of the failures while testing 

I agree it shouldn't go in just yet, thanks for sharing

Reply:
Since this is on the radar but there's no obvious solution and I'd like to get more working recipes out so I think we can move this forward